compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -39,13 +39,14 @@ object QuotesImpl {
 class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler, QuoteMatching:
 
   private val xCheckMacro: Boolean = ctx.settings.XcheckMacros.value
+  private val quoteMatcher: QuoteMatcher = QuoteMatcher(ctx.settings.YdebugMacros.value)
 
   extension [T](self: scala.quoted.Expr[T])
     def show: String =
       reflect.Printer.TreeCode.show(reflect.asTerm(self))
 
     def matches(that: scala.quoted.Expr[Any]): Boolean =
-      QuoteMatcher.treeMatch(reflect.asTerm(self), reflect.asTerm(that)).nonEmpty
+      quoteMatcher.treeMatch(reflect.asTerm(self), reflect.asTerm(that)).nonEmpty
```

Comment:
```suggestion
      QuoteMatcher(yDebugMacros).treeMatch(reflect.asTerm(self), reflect.asTerm(that)).nonEmpty
```

Reply:
For future reference, can I ask the reason for this change? Is it the case where creating `QuoteMatcher` instance for each method call is not very costful?